src/js/base/fetch.js

```diff
@@ -0,0 +1,64 @@
+//  Similar to https://github.com/akre54/Backbone.Fetch
+import $ from 'jquery';
+import { isObject, defaults, extend } from 'underscore';
+import Radio from 'backbone.radio';
+
+import { getToken } from 'js/auth';
+
+// Makes data object into `/url?param1=value1&param2=value2` string
+function getUrl(url, { data, method }) {
+  if (method !== 'GET' || !isObject(data)) return url;
+
+  const params = $.param(data);
+  if (!params) return url;
+
+  return `${ url }?${ params }`;
+}
+
+export function getData(response, dataType) {
+  if (dataType === 'json' && response.status !== 204) return response.json();
+  return response.text();
+}
+
+export function handleJSON(response) {
+  if (!response.ok) return Promise.reject(response);
+
+  return getData(response, 'json');
+}
+
+export default async(url, opts) => {
+  const token = await getToken();
+
+  const options = extend({}, opts);
+  url = getUrl(url, options);
+
+  if (options.method === 'GET') delete options.data;
```

Comment:
This allows for `fetch` to construct the get url by passing the same `{ data }` as you would on `ajax` and otherwise puts it on the body.